File: src/experiment/CheeseGrater/src/main/java/sdu/pizza/cheesegrater/PizzaOrderMessage.java
package sdu.pizza.cheesegrater;


import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data; // Using Lombok for boilerplate code
import java.util.List;

@Data // Creates getters, setters, toString, etc.
public class PizzaOrderMessage {
    @JsonProperty("pizzaId")
    private int pizzaId;

    @JsonProperty("orderId")
    private int orderId;

    @JsonProperty("orderSize")
    private int orderSize;

    @JsonProperty("startTimestamp")
    private Long startTimestamp;

    @JsonProperty("endTimestamp")
    private Long endTimestamp;

    @JsonProperty("msgDesc")
    private String msgDesc;

    @JsonProperty("sauce")
    private String sauce;

    @JsonProperty("baked")
    private boolean baked;

    @JsonProperty("cheese")
    private List<String> cheese;

    @JsonProperty("meat")
    private List<String> meat;

    @JsonProperty("veggies")
    private List<String> veggies;
}

File: src/experiment/cheese/CheeseGrater/Data/PizzaOrderMessage.cs
using System.Text.Json.Serialization;

namespace CheeseGrater.Data;

public class PizzaOrderMessage
{
    [JsonPropertyName("pizzaId")]
    public int PizzaId { get; set; }
    
    [JsonPropertyName("orderId")]
    public string OrderId { get; set; }
    
    [JsonPropertyName("orderSize")]
    public int OrderSize { get; set; }
    
    [JsonPropertyName("startTimestamp")]
    public long? StartTimestamp { get; set; }
    
    [JsonPropertyName("endTimestamp")]
    public long? EndTimestamp { get; set; }
    
    [JsonPropertyName("msgDesc")]
    public string? MsgDesc { get; set; }
    
    [JsonPropertyName("sauce")]
    public string? Sauce { get; set; }
    
    [JsonPropertyName("baked")]
    public bool Baked { get; set; }
    
    [JsonPropertyName("cheese")]
    public List<string> Cheese { get; set; } = [];
    
    [JsonPropertyName("meat")]
    public List<string> Meat { get; set; } = [];
    
    [JsonPropertyName("veggies")]
    public List<string> Veggies { get; set; } = [];
}